vm/src/obj/objbytes.rs

```diff
@@ -201,3 +203,58 @@ fn bytes_iter(obj: PyBytesRef, _vm: &VirtualMachine) -> PyIteratorValue {
         iterated_obj: obj.into_object(),
     }
 }
+
+fn compare_slice(a: &[u8], b: &[u8]) -> bool {
+    for (i, j) in a.iter().zip(b.iter()) {
+        if i != j {
+            return false;
+        }
+    }
+    return true;
+}
+
+fn compare_vec(a: &Vec<u8>, b: &Vec<u8>) -> bool {
+    let a_len = a.len();
+    let b_len = b.len();
+    for (n, i) in a.iter().enumerate() {
+        if n + b_len <= a_len && *i == b[0] {
+            if compare_slice(&a[n..n + b_len], b) {
```

Comment:
Replace this by `if &a[n..n + b_len] == &b {` if this is possible, slices should implement the `Eq` trait.